streams/integration-tests/src/test/java/org/apache/kafka/streams/integration/KafkaStreamsTelemetryIntegrationTest.java

```diff
@@ -180,15 +180,16 @@ public void shouldPushMetricsToBroker(final String recordingLevel) throws Except
                         final String name = mn.name().replace('-', '.');
                         final String group = mn.group().replace("-metrics", "").replace('-', '.');
                         return "org.apache.kafka." + group + "." + name;
-                    }).sorted().collect(Collectors.toList());
+                    }).filter(name -> !name.equals("org.apache.kafka.stream.thread.state"))// telemetry reporter filters out string metrics
+                    .sorted().collect(Collectors.toList());
```

Comment:
Telemetry filters out string metrics, so I remove it here from the expected metrics list

Reply:
I do not understand this. According to the KIP and the code in `ThreadMetrics` the type of the metric is numeric.

Reply:
We also added a  `state` metric for stream threads that reports the current state of the thread as a string.

Reply:
Got it! 
I thought `org.apache.kafka.stream.thread.thread.state` was a typo and it should actually be `org.apache.kafka.stream.thread.state`. So I was wondering, why you wanted to exclude the metric you explicitly added for sending it to the broker from the test.